src/frontend/apps/impress/src/features/docs/doc-editor/components/AIButton.tsx

```diff
@@ -212,14 +219,23 @@ const AIMenuItemTransform = ({
   icon,
 }: PropsWithChildren<AIMenuItemTransform>) => {
   const { mutateAsync: requestAI, isPending } = useDocAITransform();
+  const editor = useBlockNoteEditor();
+
+  const requestAIAction = async (selectedBlocks: Block[]) => {
+    const text = await editor.blocksToMarkdownLossy(selectedBlocks);
 
-  const requestAIAction = async (markdown: string) => {
     const responseAI = await requestAI({
-      text: markdown,
+      text,
       action,
       docId,
     });
-    return responseAI.answer;
+
+    if (!responseAI || !responseAI.answer) {
+      throw new Error('No response from AI');
+    }
+
+    const markdown = await editor.tryParseMarkdownToBlocks(responseAI.answer);
```

Comment:
Of course, non-translate AI actions like these will still be "lossy". I'll think of a blocknote-based solution for this (as part of BlockNote AI features)

Reply:
Yes, this part is still lossy, I tried but I didn't have good AI results when it is with actions, the AI has difficulty to do the request when the text is too split:
https://github.com/numerique-gouv/impress/blob/f45e724b1797693841b9304c4d7dcc6939309bad/src/frontend/apps/impress/src/features/docs/doc-editor/api/useDocAITransform.tsx#L5-L9

I think it is less likely that users will select tables or complicated structures for these actions (to monitor).